@xen-orchestra/lite/src/libs/xapi.ts

```diff
@@ -36,49 +37,58 @@ export interface ObjectsByType extends Map<string, Map<string, XapiObject>> {
   get<T extends keyof types>(key: T): Map<string, types[T]> | undefined
 }
 
-export default class XapiConnection {
+export default class XapiConnection extends EventEmitter {
   areObjectsFetched: Promise<void>
   connected: boolean
   objectsByType: ObjectsByType
   sessionId?: string
 
-  _connectSubscribers: Set<(status: boolean) => void>
-
-  _objectsSubscribers: Set<(objects: ObjectsByType) => void>
-
   _resolveObjectsFetched!: () => void
 
   _xapi?: {
     objects: EventEmitter & {
       all: { [id: string]: XapiObject }
     }
     connect(): Promise<void>
+    disconnect(): Promise<void>
+    call: (method: string, ...args: string[]) => Promise<unknown>
     _objectsFetched: Promise<void>
   }
 
   constructor() {
+    super()
+
     this.objectsByType = Map() as ObjectsByType
     this.connected = false
     this.areObjectsFetched = new Promise(resolve => {
       this._resolveObjectsFetched = resolve
     })
-    this._objectsSubscribers = new Set()
-    this._connectSubscribers = new Set()
   }
 
-  onObjects(cb: (objects: ObjectsByType) => void): () => void {
-    this._objectsSubscribers.add(cb)
-    return () => this._objectsSubscribers.delete(cb)
-  }
+  async reattachSession(url: string): Promise<void> {
+    const sessionId = Cookies.get('sessionId')
+    if (sessionId === undefined) {
+      return
+    }
 
-  onConnect(cb: (status: boolean) => void): () => void {
-    this._connectSubscribers.add(cb)
-    return () => this._connectSubscribers.delete(cb)
+    return this.connect({ url, sessionId })
   }
 
-  async connect({ url, user = 'root', password }: { url: string; user: string; password: string }): Promise<void> {
+  async connect({
+    url,
+    user = 'root',
+    password,
+    sessionId,
+    rememberMe = Cookies.get('rememberMe') === 'true' ?? false,
```

Comment:
```suggestion
    rememberMe = Cookies.get('rememberMe') === 'true',
```
Nullish coalescing operator not needed.